website/content/docs/job-specification/consul.mdx

```diff
@@ -0,0 +1,236 @@
+---
+layout: docs
+page_title: consul Block - Job Specification
+description: |-
+  The "consul" block allows the group or task to specify that it requires a token
+  from a HashiCorp Consul server. Nomad will automatically retrieve a Consul token
+  for the group or task.
+---
+
+# `consul` Block
+
+<Placement
+  groups={[
+    ['job', 'group', 'consul'],
+    ['job', 'group', 'task', 'consul'],
+  ]}
+/>
+
+The `consul` block specifies [Consul][] configuration options specific to a
+task. If specified at the `group` level, the configuration will apply to all
+tasks and services within the group unless a task has its own `consul` block.
```

Comment:
No, which makes it different from Vault unfortunately. But that's a quirk of how we used parse Vault for HCL1, IIRC.

Reply:
Ah interesting. I think other blocks are merged when specified in different levels (like `restart` and `update`) and `vault` is handled weirdly because the only struct that has a `Vault` field is `Task` 😅 

But documenting the behaviour is probably good for now. Highlighting the difference from other blocks could be handy.